### DISCUSSION TITLE
Compile ahead of time as an option

### DISCUSSION THREAD

**Discussion by bangedorrunt** [rktjmp/hotpot.nvim#81] [Ideas]
@rktjmp what is the reason why you don't put AOT compilation as an `hotpot.setup` option?
Some thing like 
```lua
require 'hotpot'.setup {
  compile = function(event, root_path, ...other_paths) ... end,
  .....
```
where:
  - `event` could be `on_save` which will compile when we save a file...
  - `root_path` is your config which convert `fnl/**/*.fnl` to `lua/**/*.fnl`
  - `other_paths` could be `ftplugins`... which convert `*.fnl` to `*.lua` inside a same folder.

**Comment by rktjmp** [comment]
I think that pushes too much of the "how" decisions into hotpot. `event` would have to be vim native events, they probably need to be more complex than just the event name but also maybe paths that might not match the root path, etc and it precludes using any uv file watchers as some people wont want that vs just events, while other will want it instead of events. You also need a way to specify files to ignore (macros) or enforce a particular file structure.

Ultimately it would just be `hotpot.make` but less flexible and I don't see the cost of writing a few autocommand/uv-watchers very high for a user, and they have much more control on how, why and when those compiles happen.

A simplified make API could be made as a separate plugin if someone wanted that had explicit rules already set. If it were popular enough perhaps an argument for mainlining it could be made.

**Comment by bangedorrunt** [comment]
@rktjmp thank for your explanation. I see your point

**Comment by rktjmp** [comment]
Discussion not relevant to v2. (It's actually all AOT now, you win!)
